features/cells/src/main/java/com/wire/android/feature/cells/ui/createfolder/CreateFolderViewModel.kt

```diff
@@ -22,66 +22,70 @@ import androidx.compose.runtime.getValue
+import com.wire.kalium.common.functional.onSuccess
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class CreateFolderViewModel @Inject constructor(
     val savedStateHandle: SavedStateHandle,
     private val createFolderUseCase: CreateFolderUseCase,
-) : ViewModel() {
+) : ActionsViewModel<CreateFolderViewModelAction>() {
 
     private val navArgs: CreateFolderScreenNavArgs = savedStateHandle.navArgs()
 
-    var createFolderState: CreateFolderState by mutableStateOf(CreateFolderState.Default)
-        private set
-
     val fileNameTextFieldState: TextFieldState = TextFieldState()
 
-    var displayNameState: DisplayNameState by mutableStateOf(DisplayNameState())
+    var viewState: CreateFolderViewState by mutableStateOf(CreateFolderViewState())
         private set
 
-    private val _isCreating = MutableStateFlow(false)
-    val isCreating = _isCreating
-
     init {
         viewModelScope.launch {
-            fileNameTextFieldState.textAsFlow().collectLatest {
-                displayNameState = displayNameState.copy(
-                    saveEnabled = it.trim().isNotEmpty() && it.length <= NAME_MAX_COUNT && !it.contains("/"),
-                    error = when {
-                        it.length > NAME_MAX_COUNT -> DisplayNameState.NameError.TextFieldError.NameExceedLimitError
-                        it.contains("/") -> DisplayNameState.NameError.TextFieldError.InvalidNameError
-                        else -> DisplayNameState.NameError.None
-                    }
+            fileNameTextFieldState.textAsFlow().map { it.trim() }.collectLatest { name ->
+                val fileValidationResult = name.validateFileName().takeIf { name.isNotEmpty() }
+                viewState = viewState.copy(
+                    saveEnabled = fileValidationResult == null && name.isNotEmpty(),
+                    error = fileValidationResult
                 )
             }
         }
     }
 
-    internal fun createFolder(folderName: String) {
-        _isCreating.value = true
-        viewModelScope.launch {
-            createFolderState = createFolderUseCase("${navArgs.uuid}/$folderName").fold(
-                { CreateFolderState.Failure },
-                { CreateFolderState.Success },
-            )
-            _isCreating.value = false
-        }
+    internal fun createFolder(folderName: String) = viewModelScope.launch {
+
+        viewState = viewState.copy(loading = true)
+
+        createFolderUseCase("${navArgs.uuid}/$folderName")
+            .onSuccess {
+                viewState = viewState.copy(loading = false)
+                sendAction(CreateFolderViewModelAction.Success)
+            }
+            .onFailure {
+                viewState = viewState.copy(loading = false)
```

Comment:
I think you can skip `viewState = viewState.copy(loading = false)` here and in `onSuccess`, line  71, as it's duplicated after that in line 78